src/main/java/com/redhat/devtools/intellij/lsp4ij/console/LSPConsoleView.java

```diff
@@ -0,0 +1,117 @@
+/*******************************************************************************
+ * Copyright (c) 2023 Red Hat Inc. and others.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License v. 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0, or the Apache License, Version 2.0
+ * which is available at https://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR Apache-2.0
+ *
+ * Contributors:
+ *     Red Hat Inc. - initial API and implementation
+ *******************************************************************************/
+package com.redhat.devtools.intellij.lsp4ij.console;
+
+import com.intellij.execution.impl.ConsoleViewImpl;
+import com.intellij.execution.impl.EditorHyperlinkSupport;
+import com.intellij.openapi.actionSystem.ActionManager;
+import com.intellij.openapi.actionSystem.AnAction;
+import com.intellij.openapi.editor.actions.ScrollToTheEndToolbarAction;
+import com.intellij.openapi.project.Project;
+import com.intellij.psi.search.GlobalSearchScope;
+import com.redhat.devtools.intellij.lsp4ij.LanguageServersRegistry;
+import com.redhat.devtools.intellij.lsp4ij.console.actions.AutoFoldingAction;
+import com.redhat.devtools.intellij.lsp4ij.console.actions.ClearThisConsoleAction;
+import com.redhat.devtools.intellij.lsp4ij.console.explorer.LanguageServerProcessTreeNode;
+import com.redhat.devtools.intellij.lsp4ij.settings.ServerTrace;
+import com.redhat.devtools.intellij.lsp4ij.settings.UserDefinedLanguageServerSettings;
+import org.jetbrains.annotations.NotNull;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import static com.redhat.devtools.intellij.lsp4ij.console.actions.AutoFoldingAction.shouldLSPTracesBeExpanded;
+
+/**
+ * Extends {@link ConsoleViewImpl} to support custom LSP folding by using [Trace
+ */
+public class LSPConsoleView extends ConsoleViewImpl {
+
+    private final LanguageServerProcessTreeNode processTreeNode;
+
+    public LSPConsoleView(@NotNull LanguageServerProcessTreeNode processTreeNode, @NotNull Project project,

```

Comment:
why not pass the serverDefinition directly instead of LanguageServerProcessTreeNode?

Reply:
fixed